src/java/org/apache/cassandra/index/sai/disk/v2/V2VectorIndexSearcher.java

```diff
@@ -243,6 +264,31 @@ private BitsOrPostingList bitsOrPostingListForKeyRange(QueryContext context, Abs
         }
     }
 
+    private int[] findTopApproximatePostings(int limit, int nRows, IntStream segmentRowIdsStream, NodeSimilarity.ApproximateScoreFunction scoreFunction) throws IOException
+    {
+        ArrayList<Pair<Integer, Float>> pairs = new ArrayList<>(nRows);
```

Comment:
you can use jvector SearchResult.NodeScore class if you don't want to duplicate it (will avoid boxing the int + float)